File: app/src/app/types/submission.ts
import { Timestamp } from "firebase/firestore";

export interface SubmissionData {
    sheets: number;
    publications: number;
    videos: number;
    hours: number;
    visits: number;
    studies: number;
}

export interface Submission {
    date: Timestamp;
    all: SubmissionData;
    publishers: SubmissionData;
    auxilaryPioneers: SubmissionData;
    regularPionners: SubmissionData;
}

File: rewarded-server/entities/submission.go
package entities

import "time"

type SubmissionData struct {
  Sheets  int `json:"sheets" bson:"sheets,omitempty"`
  Hours   int `json:"hours" bson:"hours,omitempty"`
  Studies int `json:"studies" bson:"studies,omitempty"`
}

type Submission struct {
  Date              time.Time      `bson:"date" bson:",omitempty"`
  All               SubmissionData `bson:"all" bson:"all,omitempty"`
  publishers        SubmissionData `bson:"publishers" bson:"publishers,omitempty"`
  AuxiliaryPioneers SubmissionData `json:"auxiliaryPioneers" bson:"auxiliaryPioneers,omitempty"`
  RegularPioneers   SubmissionData `json:"regularPioneers" bson:"regularPioneers,omitempty"`
  RealmId           string         `json:"realmId" bson:"realmId,omitempty"`
}
